### DISCUSSION TITLE
What is the use of path for socket.io, how is that different from custom namespace

### DISCUSSION THREAD

**Discussion by Zeus-Iqd** [socketio/socket.io#4679] [Q&A] [answered]
const socket = io('http://myserver/admin', {
  path: '/mypath'
});
[socket.io doc](https://socket.io/docs/client-api/) said with that code the socket connects to the admin namespace, with the custom path mypath. But what the use of path here? Can't namespace alone do the job?

Or under what circumstance do I need to set custom path?

**Comment by darrachequesne** [comment, marked as answer]
Hi! The `path` option refers to the request path. With `path: '/mypath'`, you will see in your network panel HTTP requests like:

```
GET http://myserver/mypath/?EIO=4&...'
```

The `path` option defaults to `/socket.io/`:

```
GET http://myserver/socket.io/?EIO=4&...'
```

Reference: https://socket.io/docs/v4/client-options/#path

On the other hand, the namespace (the `/admin` in your example) is the part of your application you'd like to connect to. It provides a way to split your application logic into independent modules, while using the same WebSocket connection under the hood (multiplexing).

Reference: https://socket.io/docs/v4/namespaces/

With `io('http://myserver/admin')`, you reach to `/admin` namespace, with `io('http://myserver')` you reach the main namespace. The data will not be found in the request URL, but in the payload.

Reference: https://socket.io/docs/v4/socket-io-protocol/#format
